src/jshint.js

```diff
@@ -3754,18 +3754,25 @@ var JSHINT = (function() {
     var b;
 
     function doCatch() {
-      var e;
+      var params = [];
 
       advance("catch");
       advance("(");
 
-      e = state.tokens.next;
-      if (e.type !== "(identifier)") {
-        warning("E030", state.tokens.next, e.value);
-        e = null;
+      if (checkPunctuators(state.tokens.next, ["[", "{"])) {
+        var tokens = destructuringExpression();
+        var t;
+        for (t in tokens) {
+          t = tokens[t];
+          if (t.id) {
+            params.push({ id: t.id, token: t });
+          }
+        }
+      } else if (state.tokens.next.type !== "(identifier)") {
+        warning("E030", state.tokens.next, state.tokens.next.value);
       } else {
         // only advance if we have an identifier so we can continue parsing in the most common error - that no param is given.
-        advance();
+        params.push({ id: identifier(), token: state.tokens.curr, type: "exception" });
```

Comment:
Sorry I missed you were specifically guarding against it before going into the function